007-design-docs.md

```diff
@@ -0,0 +1,94 @@
+
+A design document would be split into the following parts:
+
+**Title**
+
+The title of the design document, normally this would pertain to a single component or one larger aspect of the whole of Strimzi. New pages inclusion (but not contents) would be discussed and decided on as part of a proposal if applicable, or during development if the complexity of a component merits a design document as judged by the maintainers.
+
+**Overview, Context and Description**
+
+A high level summary that any user of Strimzi should be able understand and use to decide if it’s useful for them to read the rest of the doc.
+A description of the problem the component/code tackles, why the program/code is necessary, what people need to know to assess the success of the implementation.
+
+**Configuration**
+Describe how users interact with this component or system. I.e. configured via envars, custom resources, REST calls...
+This information will be invaluable to a reader to understand where the configuration points are and how to interact with the system/component.
+
+**Dependencies and Scope**
+Is this intended to be a standalone component, are other components dependent on it, or is it dependent on other components.
+Is this intended for a single Kafka cluster or to interact with multiple.
+
+
+### When is a design document complete
+
+A design document is considered complete when the reviewers and maintainers believe the document to accurately reflect the implementation of the components behaviour. 
+Once approved the document is merged and is understood to be correct, any mistakes or changes to behaviour should be corrected in the form of a pull request alongside the code that changes the behavior.
+
+A lot of implementation issues, problems and compromises are only clear during or after development of a feature or change, so a design doc is recommended to be written alongside the code development, this helps in avoiding the writing of invalid design documents and help hilight potentially problematic elements of implementation or important holes in the testing strategy. 
+
+Having the design document within the pull request allows contributors and reviewers to evolve the design while developing and during the review. This allows tweaks to the design, for current discussion and future reading, as well as to the code accordingly. This ensures the code and design docs evolve in parallel.
+
+
+## Summary of changes
+
+In an effort to improve where we are, the above guidelines are suggestions for future contributors but no enforced or mandated. This means that we will not place the entire burden on the first person to come along who wants to make a change in a given area. 
```

Comment:
"no enforced" -> "not enforced"

It might be clearer to simplify this, it seems like if it isn't enforced then design docs will never be written. Maybe just saying "In an effort to improve where we are, the above guidelines are suggested for future contributions that are either new components or significant changes to the design or behaviour of existing components. This could result in some components only being partially documented but over time design documents for existing components can be contributed."

Then I would be tempted to get rid of the rest of this section as you are sort of repeating yourself.